Find a functionally equivalent code:android.bluetooth.BluetoothGatt.executeReliableWrite:COMMENT
Method Modifier: public      
Comment:/**
 * Executes a reliable write transaction for a given remote device.
 *
 * <p>This function will commit all queued up characteristic write
 * operations for a given remote device.
 *
 * <p>A {@link BluetoothGattCallback#onReliableWriteCompleted} callback is
 * invoked to indicate whether the transaction has been executed correctly.
 *
 * <p>Requires {@link android.Manifest.permission#BLUETOOTH} permission.
 *
 * @return true, if the request to execute the transaction has been sent
 */

@@ -1,12 +1,18 @@
 {
-    if (DBG)
+    if (VDBG)
         Log.d(TAG, "executeReliableWrite() - device: " + mDevice.getAddress());
     if (mService == null || mClientIf == 0)
         return false;
+    synchronized (mDeviceBusy) {
+        if (mDeviceBusy)
+            return false;
+        mDeviceBusy = true;
+    }
     try {
         mService.endReliableWrite(mClientIf, mDevice.getAddress(), true);
     } catch (RemoteException e) {
         Log.e(TAG, "", e);
+        mDeviceBusy = false;
         return false;
     }
     return true;

